### PR TITLE
Create technical-analysis.md

### PR DESCRIPTION
**Pull request by krtaylor**
Had to cut/paste and type this in by hand, all self-formatted in markdown. Please proof carefully against: https://confidentialcomputing.io/wp-content/uploads/sites/10/2023/03/CCC-A-Technical-Analysis-of-Confidential-Computing-v1.3_unlocked.pdf

### CODE REVIEW

**Comment by dcmiddle** [comment]
Thanks again for the big lift to do this by hand @krtaylor 
@confidential-computing/tac-voting-members let's get some reviews on this.
Please compare the text against the published version on the web page. Layout will undoubtedly be different and that is not a goal right now. Just the accuracy of the text.

File: analysis/technical-analysis.md [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+approaches that do not use a hardware-based TEE.
+
+For more discussion of TEE definitions used in the industry, see the discussion at [6].
+
+### 4 Related Technologies
+
+The TAC conducted a survey of various terms in the industry related to protecting data in use, and
+composed the following Venn diagram of technologies:
+
+![Alt text](related-terminology.png "Related Terminology")
```
Comment (dthaler) [inline analysis/technical-analysis.md:140]:
```suggestion
![Alt text](related-terminology.png "Related Technologies")
```

File: analysis/technical-analysis.md [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+
+Figure 1 - Venn diagram of various technologies used to protect data in use
+
+Unfortunately, unlike the term “confidential computing”, several of the terms used in the diagram have
+multiple competing definitions. For example, “privacy-preserving computation” is variously defined
+as being synonymous with multi-party computation, or covering both multi-party computation and
+homomorphic encryption, or even (e.g., in [2]) covering the entire space of protecting data in use.
+This diagram illustrates why we refer to confidential computing as protecting data in use by using a
```
Comment (dthaler) [inline analysis/technical-analysis.md:148]:
```suggestion

This diagram illustrates why we refer to confidential computing as protecting data in use by using a
```

File: analysis/technical-analysis.md [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+hardware-based TEE, to distinguish it from other techniques.
+
+**4.1 Security Comparisons**
+
+The following table compares a typical TEE implementation with typical implementations of two other
+emerging classes of solution that protect data in use, Homomorphic Encryption (HE) and Trusted Platform
+Modules (TPM).
```
Comment (dthaler) [inline analysis/technical-analysis.md:155]: Homomorphic Encryption and Trusted Platform Modules should both be links, like they are in the whitepaper

File: analysis/technical-analysis.md [hunk 4]
```diff
@@ -0,0 +1,471 @@
...
+trusted state at the time that the evidence was generated. For a longer discussion including other variations, see [4].
+
+Attestation protocol designs and implementations must also provide assurances around specific security
+properties, including:
+
+* **Unforgeability**: Adversaries cannot produce a signature on a message that links to a trusted hardware
+component’s signature provided that the trusted hardware component never signed this message.
+* **Revocation**: If a trusted hardware component is compromised, signatures from compromised keys are
+no longer accepted.
+Some attestation schemes also provide:
```
Comment (dthaler) [inline analysis/technical-analysis.md:395]:
```suggestion

Some attestation schemes also provide:
```

**Review by ericvoit — APPROVED** [review]

**Review by ericvoit — APPROVED** [review]

File: analysis/technical-analysis.md [hunk 5]
```diff
@@ -0,0 +1,471 @@
...
+### 6 Attestation
+
+Attestation is the process by which one party, called a “Verifier”, assesses the trustworthiness of a potentially
+untrusted peer, i.e., the “Attester”. (These terms are used consistent with the Internet Engineering Task
+Force’s “Remote Attestation Procedures Architecture”[4].) The goal of attestation is to allow the Verifier
+to gain confidence in the trustworthiness of the Attester by obtaining an authentic, accurate, and timely
+report about the software and data state of the Attester.
+
+**6.1 Hardware-Based Attestation**
+Hardware-based attestation schemes rely on a trusted hardware component and associated firmware to execute
```
Comment (dcmiddle) [inline analysis/technical-analysis.md:370]: missing line break
```suggestion

Hardware-based attestation schemes rely on a trusted hardware component and associated firmware to execute
```

File: analysis/technical-analysis.md [hunk 6]
```diff
@@ -0,0 +1,474 @@
...
+homomorphic encryption, or even (e.g., in [2]) covering the entire space of protecting data in use.
+
+This diagram illustrates why we refer to confidential computing as protecting data in use by using a
+hardware-based TEE, to distinguish it from other techniques.
+
+**4.1 Security Comparisons**
+
+The following table compares a typical TEE implementation with typical implementations of two other
+emerging classes of solution that protect data in use, Homomorphic Encryption (HE) and Trusted Platform
+Modules (TPM).
```
Comment (dcmiddle) [inline analysis/technical-analysis.md:156]:
```suggestion
emerging classes of solution that protect data in use, [Homomorphic Encryption](https://homomorphicencryption.org/) (HE) 
and [Trusted Platform Modules](https://trustedcomputinggroup.org/) (TPM).
```

**Review by dcmiddle — APPROVED** [review]

File: analysis/technical-analysis.md [hunk 7]
```diff
@@ -0,0 +1,474 @@
...
+approaches that do not use a hardware-based TEE.
+
+For more discussion of TEE definitions used in the industry, see the discussion at [6].
+
+### 4 Related Technologies
+
+The TAC conducted a survey of various terms in the industry related to protecting data in use, and
+composed the following Venn diagram of technologies:
+
+![Alt text](related-terminology.png "Related Technologies")
```
Comment (thomas-fossati) [inline analysis/technical-analysis.md:140]: This will not render correctly since the PNG file is not in the folder.

(The name of the PNG should probably be `related-technologies.png`.)

File: analysis/technical-analysis.md [hunk 8]
```diff
@@ -0,0 +1,474 @@
...
+to directly observe the data as it is being derived. However, an attacker may be able to use side-channels
+to determine the value of *n*.
+
+One side channel an attacker could potentially exploit if they have physical access to the machine is to
+accurately measure the power usage of the TEE CPU during method execution. For example, if the TEE CPU
+has a different power usage for addition compared with doubling, the attacker might be able to derive *n*
+from the power profile. Not all side channel attacks require physical access, some can be achieved from
+software by measuring the time taken to perform contrived operations.
+
+![Alt text](attack-using-power.png "Attack using Power Values")
```
Comment (thomas-fossati) [inline analysis/technical-analysis.md:324]: Ditto, the linked image is not present.

File: analysis/technical-analysis.md [hunk 9]
```diff
@@ -0,0 +1,474 @@
...
+concern across the technology and security community.
+
+**5.3.2 Example**
+
+Here we give a simplified example from elliptic curve cryptography (ECC). ECC uses elliptic curve point
+multiplication to produce a one-way function. In ECC, multiplying a point P on a curve by *n* results in a new
+point on the curve, *Q*. ECC relies on the fact that it is intractable to determine *n* given both *P* and *Q*. A TEE
+might implement this multiplication using a ‘double-and-add’ method to calculate *Q*. In simplified terms,
+the method iterates through the binary representation of *n* one bit at a time and performs a *doubling*
+operation for each *0* bit and a *doubling* and *addition* operation for each *1* bit.
```
Comment (thomas-fossati) [inline analysis/technical-analysis.md:312]: Just a note: GitHub supports [MathJax](https://www.mathjax.org), so we could get fancier rendering using LaTeX style `$P$` instead of `*P*`.

**Review by thomas-fossati — COMMENTED** [review]
LGTM, thanks!

A couple of comments inlined.
